service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/TransferSignedUrlsToGcsBucketStep.java

```diff
@@ -0,0 +1,109 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.bucket;
+
+import static bio.terra.workspace.service.resource.controlled.flight.clone.bucket.StorageTransferServiceUtils.createTransferJob;
+import static bio.terra.workspace.service.resource.controlled.flight.clone.bucket.StorageTransferServiceUtils.createTransferSpecForSignedUrl;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.storagetransfer.v1.Storagetransfer;
+import com.google.api.services.storagetransfer.v1.model.TransferJob;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TransferSignedUrlsToGcsBucketStep implements Step {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(TransferSignedUrlsToGcsBucketStep.class);
+  private final Storagetransfer storagetransfer;
+
+  public TransferSignedUrlsToGcsBucketStep(Storagetransfer storageTransfer) {
+    this.storagetransfer = storageTransfer;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FlightMap workingMap = context.getWorkingMap();
+    FlightUtils.validateRequiredEntries(
+        workingMap,
+        ControlledResourceKeys.CONTROL_PLANE_PROJECT_ID,
+        ControlledResourceKeys.STORAGE_TRANSFER_SERVICE_SA_EMAIL);
+    FlightUtils.validateRequiredEntries(
+        context.getInputParameters(), ControlledResourceKeys.SIGNED_URL_LIST);
+
+    String transferJobName =
+        StorageTransferServiceUtils.createTransferJobName(context.getFlightId());
+    workingMap.put(ControlledResourceKeys.STORAGE_TRANSFER_JOB_NAME, transferJobName);
+
+    String controlPlaneProjectId =
+        workingMap.get(ControlledResourceKeys.CONTROL_PLANE_PROJECT_ID, String.class);
+
+    LOGGER.info(
+        "Creating transfer job named {} in project {}", transferJobName, controlPlaneProjectId);
+
+    String signedUrlList =
+        context.getInputParameters().get(ControlledResourceKeys.SIGNED_URL_LIST, String.class);
+
+    var destinationBucketName =
+        context
+            .getInputParameters()
+            .get(ControlledResourceKeys.DESTINATION_BUCKET_NAME_FOR_SIGNED_URL_LIST, String.class);
+
+    // Look up the transfer job by name. If it's found, it means we are restarting this step and
```

Comment:
I think this is a common code fragment. Not sure if it is reasonable, but maybe it could be shared in utils?